test/test_deltafilewrapper.cpp

```diff
@@ -970,7 +976,7 @@ TEST_CASE( "Delta File Wrapper" )
             },
             "old": {
               "attributes": {
-                "attachment": null,
+                "attachment": %3,
```

Comment:
Why is that? I think it is misleading if the value is `""` instead of pure JSON `null`? Probably needs fix upstream or there is some hidden arguments I don't understand.

Reply:
I suspect the delta code will need updating to take this new truth into account:

In Qt5, QVariant(QString()).isNull() is true
In Qt6, that now returns false, because while the qstring is null the qvariant itself isn't

Does that make you think of any part of the delta code? :)

Reply:
hm, it seems is caused from `oldVal.isNull()` not returning true, therefore `QJsonValue::fromVariant( oldVal ) )` is returning `""`.
It is here and https://github.com/opengisch/QField/blob/master/src/core/deltafilewrapper.cpp#L585-L618

Reply:
@suricactus , I suspect we'll have to use QgsVariantUtils::isNull (https://github.com/qgis/QGIS/commit/8171b76ddd64049715bdd3229ceeaf8ac8dd73b9) when we update to QGIS 3.28, that'll be the proper fix.

Reply:
Qt6 builds already build against qgis 3.28

Reply:
@m-kuhn , true that!

Reply:
@suricactus , @m-kuhn , fix incoming.

Reply:
if you don't mind, I will push a change that covers all the places I used this logic.

Reply:
@suricactus , that'd be fantastic, you're level of intimacy makes you a better committer for this :) you can close the PR I opened.